package/src/components/MessageList/MessageList.tsx

```diff
@@ -108,6 +108,36 @@ const flatListViewabilityConfig: ViewabilityConfig = {
   viewAreaCoveragePercentThreshold: 1,
 };
 
+const hasReadLastMessage = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+>(
+  channel: Channel<StreamChatGenerics>,
+  userId: string,
+) => {
+  const latestMessageIdInChannel = channel.state.latestMessages.slice(-1)[0]?.id;
```

Comment:
This can potentially fail if there are no messages in the `channel` state, could you please try that case ?

Reply:
Good point I will check

Reply:
It doesn't. Just tested it now. When the array is empty the `channel.state.latestMessages.slice(-1)[0]` would return undefined and ?.id on it will also be undefined.